internal/cgroups/cgroups.go

```diff
@@ -115,3 +138,67 @@ func (cg CGroups) CPUQuota() (float64, bool, error) {
 
 	return float64(cfsQuotaUs) / float64(cfsPeriodUs), true, nil
 }
+
+// IsCGroupV2 returns true if the system supports and uses cgroup2.
+// It gets the required information for deciding from mountinfo file.
+func IsCGroupV2() (bool, error) {
+	return isCGroupV2(_procPathMountInfo)
+}
+
+func isCGroupV2(procPathMountInfo string) (bool, error) {
+	isV2 := false
+	newMountPoint := func(mp *MountPoint) error {
+		if mp.FSType == _cgroupv2FSType && mp.MountPoint == _cgroupv2MountPoint {
+			isV2 = true
+		}
+		return nil
+	}
+	if err := parseMountInfo(procPathMountInfo, newMountPoint); err != nil {
+		return false, err
+	}
+	if isV2 {
+		return true, nil
+	}
+	return false, nil
+}
+
+// CPUQuotaV2 returns the CPU quota applied with the CPU cgroup2 controller.
+// It is a result of reading cpu quota and period from cpu.max file.
+// It will return `cpu.max / cpu.period`. If cpu.max is set to max, it returns
+// (-1, false, nil)
+func CPUQuotaV2() (float64, bool, error) {
+	return cpuQuotaV2(_cgroupv2MountPoint, _cgroupv2CPUMax)
+}
+
+func cpuQuotaV2(cgroupv2MountPoint, cgroupv2CPUMax string) (float64, bool, error) {
+	cpuMaxParams, err := os.Open(path.Join(cgroupv2MountPoint, cgroupv2CPUMax))
+	if err != nil {
+		if os.IsNotExist(err) {
+			return -1, false, nil
+		}
+		return -1, false, err
+	}
+	scanner := bufio.NewScanner(cpuMaxParams)
+	if scanner.Scan() {
+		fields := strings.Fields(scanner.Text())
+		if len(fields) != 2 {
+			return -1, false, fmt.Errorf("invalid format")
+		}
```

Comment:
```
  cpu.max
	A read-write two value file which exists on non-root cgroups.
	The default is "max 100000".

	The maximum bandwidth limit.  It's in the following format::

	  $MAX $PERIOD

	which indicates that the group may consume upto $MAX in each
	$PERIOD duration.  "max" for $MAX indicates no limit.  If only
	one number is written, $MAX is updated.
```

https://www.kernel.org/doc/Documentation/admin-guide/cgroup-v2.rst

I think it's safe to assume that if this format were to change the semantics of it would be different, hence this code would need to be revised to support it.

Reply:
Thanks for the comments. I've changed the code accordingly.